TypeInjections/Tool.Test/ToolTests.fs

```diff
@@ -0,0 +1,148 @@
+        Tool.checkoutCommit "0f836ae9a38334a7def8d5f3c1ff5941ab0cc109" "https://github.com/tschuerl/DafnyExample.git" $"{resourcePath}/Git/LongHashTest/Output/Old"
+        cmpFiles $"{resourcePath}/Git/LongHashTest/Expected" $"{resourcePath}/Git/LongHashTest/Output"
+
+
+    [<Test>]
+    // Can we clone remote repo based on short hash (first 7 characters)
+    let GitShortHashTest () =
+        // clear out old output
+        if Directory.Exists($"{resourcePath}/Git/ShortHashTest/Output") then
+            Directory.Delete($"{resourcePath}/Git/ShortHashTest/Output", true)
+
+        Tool.checkoutCommit "12e2759" "https://github.com/tschuerl/DafnyExample.git" $"{resourcePath}/Git/ShortHashTest/Output/New"
+        Tool.checkoutCommit "0f836a" "https://github.com/tschuerl/DafnyExample.git" $"{resourcePath}/Git/ShortHashTest/Output/Old"
+        cmpFiles $"{resourcePath}/Git/ShortHashTest/Expected" $"{resourcePath}/Git/ShortHashTest/Output"
+
+
+    [<Test>]
+    // can we get commits from dif branches
+    let GitDifBranchHashTest () =
+        // clear out old output
+        if Directory.Exists($"{resourcePath}/Git/DifBranchHashTest/Output") then
+            Directory.Delete($"{resourcePath}/Git/DifBranchHashTest/Output", true)
+            
+        Tool.checkoutCommit "0f836ae" "https://github.com/tschuerl/DafnyExample.git" $"{resourcePath}/Git/DifBranchHashTest/Output/Old"
+        Tool.checkoutCommit "6734af6" "https://github.com/tschuerl/DafnyExample.git" $"{resourcePath}/Git/DifBranchHashTest/Output/New"
+        cmpFiles $"{resourcePath}/Git/DifBranchHashTest/Expected" $"{resourcePath}/Git/DifBranchHashTest/Output"
+
+    [<Test>]
+    // raise exception for bad git URL
+    let GitBadURLTest () =
+        Assert.Catch
+            (fun () ->
+                Tool.checkoutCommit "0f836ae" "https://github.com/tschuerl/IMAGINARY_REPO.git" $"{resourcePath}/Git/BadURLTest/Output")
+        |> ignore
+
+
+    [<Test>]
+    // no repository / bad repository in directory
+    let GitBadLocation () =
+        // no git repo
+        let exn =
+            Assert.Catch(fun () -> Tool.checkoutCommit "0f836ae" $"{resourcePath}/Git/BadLocation" $"{resourcePath}/Git/BadLocation/Generated")
+
+        Console.WriteLine $"Exn was: {exn.Message}"
+        // git repo has no dafny files
+        
+        (* TODO ask Matthias how to replicate local git repo in unit testing (git wanted me to make a submodule)
+        let outFolder = $"{resourcePath}/Git/BadLocation/Generated/Output"
+        // ensure output directory is empty (in Tool.fs, other method handles this, checkout needs this to be done manually)
+        if Directory.Exists(outFolder) then
+            Directory.Delete(outFolder, true)
+
+        let exnNoDafny =
+            Assert.Catch(fun () -> Tool.checkoutCommit "20e447" $"{resourcePath}/Git/BadLocation/RepoNoDafny" outFolder)
+
+        Console.WriteLine $"Exn was: {exnNoDafny.Message}" *)
```

Comment:
I know we don't comment code, but I wanted to ask you about testing `typecart git` with a local repository. I made a repository inside `Tool/Resources` but when I tried to commit a change, I got an error. The error message wanted me to add a `git submodule`

Reply:
I think we'd need to mock the git repository, so the library thinks there is one, but in reality it's just a mock. However, I think we can defer those tests for now. Just add a comment `// TODO: test typecart git --clone <URL>`

Reply:
BTW: the same is true for all the git commands above, we should not test these with real git repositories. Please add a TODO instead of using your hard-coded personal reposiotry. We can look into it later.

This might be relevant, if we have time to get back to it:
https://github.com/libgit2/libgit2sharp/blob/master/LibGit2Sharp.Tests/MockingFixture.cs